src/functions.js

```diff
@@ -430,11 +504,35 @@ var _ = Mavo.Functions = {
 	uppercase: text => str(text).toUpperCase(),
 	lowercase: text => str(text).toLowerCase(),
 
-	from: (haystack, needle) => _.between(haystack, needle),
-	fromlast: (haystack, needle) => _.between(haystack, needle, "", true),
-	to: (haystack, needle) => _.between(haystack, "", needle),
-	tofirst: (haystack, needle) => _.between(haystack, "", needle, true),
+	from: (haystack, needle) => {
+		return Mavo.Script.binaryOperation(haystack, needle, {
+			scalar: (haystack, needle) => _.between(haystack, needle),
+			identity: null
+		});
+	},
+	
+	fromlast: (haystack, needle) => {
+		return Mavo.Script.binaryOperation(haystack, needle, {
+			scalar: (haystack, needle) => _.between(haystack, needle, "", true),
+			identity: null
+		});
+	},
 
+	to: (haystack, needle) => {
+		return Mavo.Script.binaryOperation(haystack, needle, {
+			scalar: (haystack, needle) => _.between(haystack, "", needle),
+			identity: null
+		});
+	},
+
+	tofirst: (haystack, needle) => {
+		return Mavo.Script.binaryOperation(haystack, needle, {
+			scalar: (haystack, needle) => _.between(haystack, "", needle, true),
+			identity: null
+		});
+
+	},
```

Comment:
Yeah these are very repetitive, and it makes it hard for a human to process the code, as they need to mentally ignore the wrapping code around it. 

Also you seem to have missed a few, e.g. what about `lowercase` and `uppercase` (which are unary)?

I would leave the functions as they were, and do the wrapping in a separate loop afterwards, just like you did for `Math`, and use the right function (binaryOperation vs unaryOperation) based on the number of arguments.

Even functions with more than 2 arguments can be converted (e.g. `between`), and only allow for array(s) for 1-2 arguments.

Reply:
I'm implementing the new loop and I've encountered a few issues:

1. Some functions with default parameters are misinterpreted as having only one argument. Do you have any idea on how I could filter them out?
2. Some functions that already accept an array argument (e.g. `sum`, `average`) are also been misinterpreted as element-wise operations after wrapping.
3. binaryOperation functions may be a little sophisticated to wrapped with a loop since they have varying identities and unaries.

I appreciate your input! :)

Reply:
It looks like we need to include some metadata with each function, and leave the autodetection as a fallback, for those missing the metadata. 

so, in the place where the functions are defined, instead of:

```js
function_name: arg => { // code },
```

we'd do:

```js
function_name: $.extend(arg => { // code }, {meta1: value1, meta2, value2, ...}),
```

It's worth spending some time thinking what's the right metadata to include that facilitates programmer convenience (since it's not just about Mavo, but also plugin authors), i.e. that helps us figure out the right information, with the minimal amount of metadata.

Here are some options I see:
- Unary or binary? The problem is what if you want e.g. argument 1 and argument 3 to be the operation operands?
- Which arguments (by index) should be the operands? Then unary vs binary can be figured out by how many they are.
- Optionally, also properties for operand defaults and any other properties (e.g. `leftUnary` if we wish to define one). Btw you may want to send a separate PR for converting `identity` to `default` before we add even more `identity` in the codebase.
- Should the wrapping be opt in? I.e. any functions without any of this metadata would not be wrapped. This certainly prevents false wrapping of aggregates, but it makes things a little more tedious, since every single function now needs metadata. Alternatively, there could be a blacklist of functions that would not be wrapped. Or metadata that means "this function should not be wrapped automatically" (it can still be wrapped manually if we wish to do so, e.g. if it needs something special). I think the best course of action depends on how many we have that need wrapping and how many that don't. Can you count them?

Reply:
There are *approximately* **7 unary operations** and **15 binary operations** (including 2 binary Math functions) to be potentially wrapped.

I put "approximately" because there are some that have more than 2 arguments but could potentially still work, like the `between` you mentioned.

Reply:
But how many should **not** be wrapped?

Reply:
If we are not including multi-arg functions at all, then about 22 functions shouldn't be wrapped (approx. 44 functions in total). But if we *are* wrapping *some* multi-arg *and* aggregate functions, then the number will be a lot less, but still somewhere around 10 will be unwrapped, I'm guessing?

Regarding the options you listed, I like the ideas in the second and third bullets. As for learnability & readability for new contributors, I feel like as long as we document how the metadata thing works (comments and/or README) it will be more advantageous in the long run.